include/inexor/vulkan-renderer/exception.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <vulkan/vulkan_core.h>
+
+#include <stdexcept>
+
+namespace inexor::vulkan_renderer {
+
+/// @brief A custom exception class for handling Vulkan API exceptions.
+class Exception : public std::exception {
+
+    // The return value which was given by the failed vkFunction call.
+    const VkResult m_result;
+    // The input error message which is given by the code passage which throws the exception.
+    const std::string m_error_message;
```

Comment:
Does this really need to be a member? It looks like it's only used in the constructor to construct the final message.

Reply:
Correct.

Reply:
Done.